### DISCUSSION TITLE
kibana-discover-version 8.8 not found in enum with elastalert2 version 2.12

### DISCUSSION THREAD

**Discussion by alexAtQuanos** [jertel/elastalert2#1211] [Q&A] [answered]
Hello,

Im using the kubernetes deployment 

`helm upgrade  --install --version="2.12.0" elastalert2  elastalert2/elastalert2 --namespace $KUBE_NAMESPACE`

I tried to use kibana-discover-version 8.8 with a teams alert. I get this Error:

ERROR:elastalert:Could not load rule /opt/elastalert/rules/alerting-mail.yaml: Invalid Rule file: /opt/elastalert/rules/alerting-mail.yaml
'8.8' is not one of ['8.5', '8.4', '8.3', '8.2', '8.1', '8.0', '7.17', '7.16', '7.15', '7.14', '7.13', '7.12', '7.11', '7.10', '7.9', '7.8', '7.7', '7.6', '7.5', '7.4', '7.3', '7.2', '7.1', '7.0']

Failed validating 'enum' in schema['properties']['kibana_discover_version']:
    {'enum': ['8.5',
              '8.4',
              '8.3',
              '8.2',
              '8.1',
              '8.0',
              '7.17',
              '7.16',
              '7.15',
              '7.14',
              '7.13',
              '7.12',
              '7.11',
              '7.10',
              '7.9',
              '7.8',
              '7.7',
              '7.6',
              '7.5',
              '7.4',
              '7.3',
              '7.2',
              '7.1',
              '7.0'],
     'type': 'string'}

On instance['kibana_discover_version']:
    '8.8'

the configured rule looks something like this: 

name: alerting
type: any
index: alerting-environment
filter:
- query:
    query_string:
      query: "type: alert"
realert:
  minutes: 1
query_key: "reason"
alert:
- "ms_teams"
ms_teams_theme_color: "#6600ff"
ms_teams_webhook_url: "XXXXXXX"
generate_kibana_discover_url: True
kibana_discover_app_url: "https://XXXXX/app/discover#/"
kibana_discover_index_pattern_id: "12345"
kibana_discover_version: "8.8"
smtp_host: "XXX.com"
alert_subject: "Alerting"
alert_text: "Alert"

**Comment by jertel** [comment, marked as answer]
I've moved this to a discussion since your cluster seems to be using an old Docker image, rather than being a bug. I'm looking at the schema in GitHub and it correctly shows all the way up to 8.8 as being a supported version. Next, I ran the following to confirm that the schema.yaml file is indeed correct in the published 2.12.0 image on Docker Hub:

```
$ docker run --rm -it --entrypoint bash jertel/elastalert2:2.12.0
Unable to find image 'jertel/elastalert2:2.12.0' locally
2.12.0: Pulling from jertel/elastalert2
8b91b88d5577: Already exists 
824416e23423: Already exists 
482d64d97d4e: Already exists 
c87b3089b2ed: Already exists 
91bdacd599c6: Already exists 
2e7adb01a5cd: Pull complete 
0a550156df70: Pull complete 
4f4fb700ef54: Pull complete 
Digest: sha256:0ad9d5e46ef85cec81636bb23a91314c7c8daf41802905d2b7550bce00434b3b
Status: Downloaded newer image for jertel/elastalert2:2.12.0
elastalert@d1b53fd19d45:~$ cat /usr/local/lib/python3.11/site-packages/elastalert/schema.yaml | grep version
  kibana_discover_version: {type: string, enum: ['8.8', '8.7', '8.6', '8.5', '8.4', '8.3', '8.2', '8.1', '8.0', '7.17', '7.16', '7.15', '7.14', '7.13', '7.12', '7.11', '7.10', '7.9', '7.8', '7.7', '7.6', '7.5', '7.4', '7.3', '7.2', '7.1', '7.0']}
```

I suggest using kubectl to exec into your ElastAlert2 pod and run that same `cat` command above to confirm that your Docker image is not actually using the 2.12.0 version.

**Reply by alexAtQuanos** [reply]
You were correct. The actual image the container used is jertel/elastalert2:2.9.0
